src/module/skb/bpf/skb_hook.bpf.c

```diff
@@ -108,52 +108,19 @@ struct skb_data_ref_event {
 } __attribute__((packed));
 
 /* Must be called with a valid skb pointer */
-static __always_inline int process_skb_l2_l4(struct retis_raw_event *event,
-					     struct skb_config *cfg,
-					     struct sk_buff *skb)
+static __always_inline int process_skb_ip(struct retis_raw_event *event,
+					  struct skb_config *cfg,
+					  struct sk_buff *skb,
+					  unsigned char *head,
```

Comment:
Why do we need to pass `head` here? just to save us from BPF_CORE_READ() it again?

Reply:
I compared the resulting ebpf with and without this change. I was curious to see if the "-O2" that we're giving clang is would optimize out the extra BPF_CORE_READ() but it seems that's not the case. I was trying to avoid making the functions uglier but in this case it seems there is a small gain after all.